packages/dds/tree/src/feature-libraries/editable-tree/editableTreeContext.ts

```diff
@@ -99,14 +141,85 @@ export class ProxyContext implements EditableTreeContext {
         cursor.free();
         return proxifiedField;
     }
+
+    public attachAfterChangeHandler(afterHandler: (this: EditableTreeContext) => void): void {
+        this.afterChangeHandlers.add(afterHandler);
+    }
+
+    private handleAfterChange(): void {
+        for (const afterChangeHandler of this.afterChangeHandlers) {
+            afterChangeHandler.call(this);
```

Comment:
It seems a bit strange to take in callbacks which expect the `this` to be the context. I think it might make more sense to use functions which don't rely on `this`. That would make it possible for lambdas to be provided and access the context for example.

Generally, `this` isn't very robustly handled in typescript, and we don't use it to provide context to callbacks.

```suggestion
    public attachAfterChangeHandler(afterHandler: (context: EditableTreeContext) => void): void {
        this.afterChangeHandlers.add(afterHandler);
    }

    private handleAfterChange(): void {
        for (const afterChangeHandler of this.afterChangeHandlers) {
            afterChangeHandler(this);
```

You will also need to make a change to the declared type to match.

Reply:
Thanks for the tip. Fixed.